src/plots/linePlot.ts

```diff
@@ -349,10 +374,65 @@ export module Plots {
 
     protected _getDataToDraw() {
       let dataToDraw = new Utils.Map<Dataset, any[]> ();
-      this.datasets().forEach((dataset) => dataToDraw.set(dataset, [dataset.data()]));
+
+      this.datasets().forEach((dataset) => {
+        let data = dataset.data();
+
+        if (!this._croppedRendering) {
+          dataToDraw.set(dataset, [data]);
+          return;
+        }
+
+        let filteredDataIndices = data.map((d, i) => i);
+
+        if (this._croppedRendering) {
+          filteredDataIndices = this._filterCroppedRendering(dataset, filteredDataIndices);
+        }
```

Comment:
It will be relevant when downsampling is added. It's easier to not have a merge conflict / forget to add this if I leave it in.